Make Sparkfun Thing Plus 9160 build with secure boot

When building the B0 image as part of the secure boot, the CONFIG_GPIO=y is dropped by the build process. This causes a compile error in board.c.

**Describe the solution you'd like**
Option 1:
The setting of the power pin in the function board_sparkfun_thing_plus_nrf9160_init is disabled when CONFIG_GPIO is not defined.

Option 2:
A solution to make sure that the CONFIG_GPIO=y is set during the build processes of the secure boot.

**Additional context**
The error shows during linking, where it has an undefined reference to __device_dts_ord_XXX.

--- a/boards/arm/sparkfun_thing_plus_nrf9160/Kconfig.board
+++ b/boards/arm/sparkfun_thing_plus_nrf9160/Kconfig.board
@@ -8,8 +8,16 @@ if SOC_NRF9160_SICA
 
 config BOARD_SPARKFUN_THING_PLUS_NRF9160
 	bool "Sparkfun nRF9160 Thing Plus"
+	# The GPIO driver is required by this board's initialization code
+	# (board.c), so it is forced here to be enabled always, not only
+	# enabled by default (in defconfig).
+	select GPIO
 
 config BOARD_SPARKFUN_THING_PLUS_NRF9160_NS
 	bool "Sparkfun nRF9160 Thing Plus non-secure"
+	# The GPIO driver is required by this board's initialization code
+	# (board.c), so it is forced here to be enabled always, not only
+	# enabled by default (in defconfig).
+	select GPIO
 
 endif # SOC_NRF9160_SICA

--- a/boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig
+++ b/boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_defconfig
@@ -13,9 +13,6 @@ CONFIG_ARM_TRUSTZONE_M=y
 # Hardware stack protection
 CONFIG_HW_STACK_PROTECTION=y
 
-# enable GPIO
-CONFIG_GPIO=y
-
 # Enable uart driver
 CONFIG_SERIAL=y
 

--- a/boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_ns_defconfig
+++ b/boards/arm/sparkfun_thing_plus_nrf9160/sparkfun_thing_plus_nrf9160_ns_defconfig
@@ -16,9 +16,6 @@ CONFIG_TRUSTED_EXECUTION_NONSECURE=y
 # Hardware stack protection
 CONFIG_HW_STACK_PROTECTION=y
 
-# enable GPIO
-CONFIG_GPIO=y
-
 # Enable uart driver
 CONFIG_SERIAL=y
 

